dirsrvtests/tests/suites/webui/__init__.py

```diff
@@ -120,3 +120,113 @@ def fin():
         remove_instance_through_webui(topology_st, page, browser_name)
 
     request.addfinalizer(fin)
+
+
+def load_ldap_browser_tab(frame):
+    frame.get_by_role('tab', name='LDAP Browser', exact=True).click()
+    frame.get_by_role('button').filter(has_text='dc=example,dc=com').click()
+    frame.get_by_role('columnheader', name='Attribute').wait_for()
+    time.sleep(1)
+
+
+def create_entry(frame, entry_type, entry_data):
+    if entry_type == 'user':
+        frame.get_by_role("tabpanel", name="Tree View").get_by_role("button", name="Actions").click()
+        frame.get_by_role("menuitem", name="New ...").click()
+        frame.get_by_role("button", name="Next").click()
+        frame.get_by_role("button", name="Options menu").click()
+        frame.get_by_role("option", name="Posix Account").click()
+        frame.get_by_role("button", name="Next", exact=True).click()
+        frame.get_by_role("button", name="Next", exact=True).click()
+
+        for row in range(6):
+            frame.get_by_role("button", name=f"Place row {row} in edit mode").click()
+            if row == 0:
+                frame.get_by_role("textbox", name="_").fill(entry_data['user_name'])
+            elif row == 1:
+                frame.get_by_role("textbox", name="_").fill(entry_data['user_name'])
+            elif row == 2:
+                frame.get_by_role("textbox", name="_").fill(entry_data['user_id'])
+            elif row == 3:
+                frame.get_by_role("textbox", name="_").fill(entry_data['user_home_dir'])
+            elif row == 4:
+                frame.get_by_role("textbox", name="_").fill(entry_data['user_id'])
+            elif row == 5:
+                frame.get_by_role("textbox", name="_").fill(entry_data['user_id'])
+            frame.get_by_role("button", name=f"Save row edits for row {row}").click()
```

Comment:
I still feel that the function is just too redundant.

In my initial suggestion, I was thinking about something like that in `entry_data`:

```
entry_data = [
    {
        "type": "User",
        "name": "John Smith",
        "attrs": {
            "uid": "1204",
            "home_directory": "/user/jsmith",
            "gid": "1204",
            # etc.
            # ...
        }
    },
    {
        "type": "Group",
        "name": "testgroup"
    },
   # etc.
   # ...
    }
]
```

This way, you can get rid of all of the redundant lines and unnecessary `if`s.

Reply:
Yeah, i had feeling that this is not the right approach, but i was not able to figure out something else. Thank you for guidance. Next iteration should be better.

Reply:
What I mean is that instead of many `elif entry_type == 'group':` blocks, you can remove the duplication and use the data from the `entry_data` dict.

For example, this code is repeated over and over for every element (more or less):

```
    elif entry_type == 'group':
        frame.get_by_role("tabpanel", name="Tree View").get_by_role("button", name="Actions").click()
        frame.get_by_role("menuitem", name="New ...").click()
        frame.get_by_label("Create a new Group").check()
        frame.get_by_role("button", name="Next").click()
        ...
        frame.get_by_role("button", name="Next").click()
        frame.get_by_role("button", name="Create", exact=True).click()
        frame.get_by_role("button", name="Finish").click()
        frame.get_by_role("button").filter(has_text=f"cn={entry_data['group_name']}").wait_for()

    elif entry_type == 'ou':
        frame.get_by_role("tabpanel", name="Tree View").get_by_role("button", name="Actions").click()
        frame.get_by_role("menuitem", name="New ...").click()
        frame.get_by_label("Create a new Organizational Unit").check()
        frame.get_by_role("button", name="Next").click()
        ...
        frame.get_by_role("button", name="Next").click()
        frame.get_by_role("button", name="Create", exact=True).click()
        frame.get_by_role("button", name="Finish").click()
        frame.get_by_role("button").filter(has_text=f"ou={entry_data['ou_name']}").wait_for()
```

So it'll be much cleaner to have the lines written only once and use the data from the entry_data dict in the places where needed (i.e. `frame.get_by_label(f"Create a new {entry_data['type']}").check()`)

Reply:
I think it's possible to use  the same technic (with probably, minor modifications) that you used here:

```
for row, value in enumerate(entry_data.values()):
            if row > 5:
                break
            frame.get_by_role("button", name=f"Place row {row} in edit mode").click()
            frame.get_by_role("textbox", name="_").fill(value)
            frame.get_by_role("button", name=f"Save row edits for row {row}").click()
```

But use it generally across all objects, so you'll have no need for many `ifs` and just depend on the entry_data structure while filling in the data. Because essentially, it's repeated through all of the types (you may need to modify how entry_data is structured for checkboxes, though).

For example, for custom entry, it's very close to the example above:

```
    elif entry_type == 'custom Entry':
        frame.get_by_role("checkbox", name="Select row 0").check()
        frame.get_by_role("button", name="Next", exact=True).click()
        frame.get_by_role("checkbox", name="Select row 1").check()
        frame.get_by_role("button", name="Next", exact=True).click()
        frame.get_by_role("button", name="Place row 0 in edit mode").click()
        frame.get_by_role("textbox", name="_").fill(entry_data['uid'])
        frame.get_by_role("button", name="Save row edits for row 0").click()
        frame.get_by_role("button", name="Place row 1 in edit mode").click()
        frame.get_by_role("textbox", name="_").fill(entry_data['entry_name'])
        frame.get_by_role("button", name="Save row edits for row 1").click()
```

But okay, we spent a lot of time on this one, and it's better to progress to the next task. You can always return to this one and refactor if you'd like. :)

Still, the code looks good, and the tests pass. Ack from me.
I'll leave it for @bsimonova or @vashirov to merge if they are okay.